doc/api/events.markdown

```diff
@@ -339,7 +339,8 @@ Removes the specified `listener` from the listener array for the specified
 `removeListener` will remove, at most, one instance of a listener from the
 listener array. If any single listener has been added multiple times to the
 listener array for the specified `event`, then `removeListener` must be called
-multiple times to remove each instance.
+multiple times to remove each instance. Listeners removed from an event from
+within the emit cycle of that event will still be called in that emit cycle.  
```

Comment:
If this is a promise, I would argue that a unit test to prove that this happens would be good too.